### PR TITLE
Use less C in the RAII introduction slides

### PR DESCRIPTION
**Pull request by bernhardmgruber**
Fixes: #71

### CODE REVIEW

File: talk/morelanguage/raii.tex [hunk 1]
```diff
@@ -47,16 +47,19 @@
...
-      try {
-        FILE *handle = std::fopen(path, "w+");
-        if (nullptr == handle) { throw ... }
-        if (std::fputs(str, handle) == EOF) {
-          throw ...
-        }
-        fclose(handle);
-      } catch (...) { ... }
+      std::FILE *handle = std::fopen(path, "w+");
+      if (nullptr == handle) { throw ... }
```
Comment (hageboeck) [inline talk/morelanguage/raii.tex:51]:
```suggestion
      if (handle == nullptr) { throw ... }
```
I don't insist, but isn't this easier to read?

**Review by hageboeck — APPROVED** [review]
LGTM, with two additions:
- There's a mostly formatting-related PR to your branch that you could please merge.
- I have one code suggestion that not very important, though.

File: talk/morelanguage/raii.tex (hunk 1 above)
Comment (bernhardmgruber) [reply talk/morelanguage/raii.tex:51]: There is this clan of C++ developers that believes putting an rvalue on the LHS of an equality comparison detects accidental misspelling the operator as assignment:
```c++
if (nullptr = handle) // compiler error
if (handle = nullptr) // compiles fine (but may warn)
```
I think recent compilers warn on this anyway and you should put parentheses around it to silence the warnings:
```c++
if (handle = nullptr) // may warn
if ((handle = nullptr)) // does not warn
```

Comment (sponce) [reply talk/morelanguage/raii.tex:51]: Yes. This is the origin of this. An LHCb old convention, put in place 15-20 years before compilers added the warnings :-) Now I'm not sure it's really useful nowadays and this example should not be polluted by such considerations, so I woudl keep the change

**Pull request merged**